src/components/header.js

```diff
@@ -40,41 +40,39 @@ function Header({ slug }) {
   const [isOpen1, setIsOpen1] = useState(false);
   const [isOpen2, setIsOpen2] = useState(false);
 
-  /* first dropdown items */
-  const navDropdownLinks1 = links.navDropdown1;
-  const navDropdownLinkLabels1 = t("header.nav.dropdowns")[0]["simpleLinks"];
-  const navDropdownLength1 =
-    navDropdownLinks1.length > navDropdownLinkLabels1.length
-      ? navDropdownLinkLabels1.length
-      : navDropdownLinks1.length; // take shorter length in case there is a missing link in navDropdownLinks1 or missing label in translation file
-  const navDropdownItems1 = navDropdownLinks1.map((element, i) => {
-    return (
-      i < navDropdownLength1 && (
-        <Link to={element} key={i}>
-          {navDropdownLinkLabels1[i]}
-        </Link>
-      )
-    );
-  });
-
-  /* second dropdown items */
-  const navDropdownLinks2 = links.navDropdown2;
-  const navDropdownLinkLabels2 = t("header.nav.dropdowns")[1]["simpleLinks"];
-  const navDropdownLength2 =
-    navDropdownLinks2.length > navDropdownLinkLabels2.length
-      ? navDropdownLinkLabels2.length
-      : navDropdownLinks2.length; // take shorter length in case there is a missing link in navDropdownLinks2 or missing label in translation file
-  const navDropdownItems2 = navDropdownLinks2.map((element, i) => {
-    return (
-      i < navDropdownLength2 && (
-        <Link to={element} key={i}>
-          {navDropdownLinkLabels2[i]}
-        </Link>
-      )
-    );
+  /* dynamically creating dropdowns in navigation bar */
+  let dropdowns = [];
+  const dropdownLinks = links.navDropdowns;
+  const constFileLength = dropdownLinks.length;
+  const dropdownLabels = t("header.nav.dropdowns");
+  const translationFileLength = dropdownLabels.length;
+  const length =
+    translationFileLength > constFileLength
+      ? constFileLength
+      : translationFileLength; // take shorter length in case there is missing dropdowns in `../constants/links.js` (constants file) or `../locales/` (translation files)
+  dropdownLinks.map((_, i) => {
+    if (i < length) {
+      const navDropdownLinks = dropdownLinks[i];
+      const navDropdownLinkLabels = t("header.nav.dropdowns")[i]["simpleLinks"];
+      const navDropdownLength =
+        navDropdownLinks.length > navDropdownLinkLabels.length
+          ? navDropdownLinkLabels.length
+          : navDropdownLinks.length; // take shorter length in case there is a missing link in `../constants/links.js` (constants file) or label in  `../locales/` (translation files)
+      dropdowns.push(
```

Comment:
I think pushing into the `dropdowns` array like this might not be best practice, since you don't know which index link would be missing. If there are 10 links, and the 5th is missing, then the rest of the labels would be off by one.

This is why I was advocating for placing the link and the link label in an object in the translation.json file. Something along these lines:

```
dropdowns: [
{label: "label1", link: "/path1"},
{label: "label2", link: "/path2"}
]
```

Reply:
Check out my previous comment.

Reply:
I just edited this example - using i18next in constants  and storing the link and label translation in an array of objects?